Source/Fuse.Controls.CameraView/Android/CameraImpl.java

```diff
@@ -253,6 +257,164 @@ public void setFlashMode(String flashMode) {
+                    }
+
+                    if (parameters.isAutoWhiteBalanceLockSupported()) {
+
+                        parameters.setAutoWhiteBalanceLock(false);
+                    }
+
+                    _camera.setParameters(parameters);
+                }
+
+                int focus_area_size = 300;
+                int left = clamp(Float.valueOf((float)(x / cameraWidth) * 2000 - 1000).intValue(), focus_area_size);
+                int top = clamp(Float.valueOf((float)(y / cameraHeight) * 2000 - 1000).intValue(), focus_area_size);
+
+
+                Display display = com.fuse.Activity.getRootActivity().getWindowManager().getDefaultDisplay();
+                Double tmp;
+                Rect rect;
+                switch(display.getRotation()) {
+                    case 0: //portrait
+                        rect = new Rect(left, top, left + focus_area_size, top + focus_area_size);
+                        break;
+                    case 3: //landscape right
+                        rect = new Rect(top, left, top + focus_area_size, left + focus_area_size);
+                        break;
+                    case 2: //portrait upsidedown
+                        rect = new Rect(top, left, top + focus_area_size, left + focus_area_size);
+                        break;
+                    case 1: //landscape left
+                        rect = new Rect(left, top, left + focus_area_size, top + focus_area_size);
+                        break;
+                    default:
+                        rect = new Rect(left, top, left + focus_area_size, top + focus_area_size);
+                        break;
+                }
+
+
+                List<Camera.Area> meteringAreas = new ArrayList<Camera.Area>();
+                meteringAreas.add(new Camera.Area(rect, 1000));
+                parameters.setFocusAreas(meteringAreas);
+
+                if (focusModes.contains(Camera.Parameters.FOCUS_MODE_AUTO)) {
+                    parameters.setFocusMode(Camera.Parameters.FOCUS_MODE_AUTO);
+                }
+
+                if (parameters.getMinExposureCompensation() != 0 
+                    && parameters.getMaxExposureCompensation() != 0
+                    && parameters.isAutoExposureLockSupported()
+                ) {
+                    
+                    int exposureAmount = (parameters.getMinExposureCompensation() - parameters.getMaxExposureCompensation()) * -1;
+                    int designAdjustment = 1; //added 1 to account for a control panel covered area on the bottom of the camera that can't be tapped on for focus
+                    if ((y/cameraHeight) < 0.5) { //apply only to upper half of camera screen
+                        designAdjustment = 0;
+                    }
+
+                    int amountApplied = (int)Math.round( (exposureAmount * (y/cameraHeight)) ) + designAdjustment; 
+                    amountApplied = (amountApplied > exposureAmount) ? exposureAmount : amountApplied;
```

Comment:
Is this code based on some algorithm worth noting? I'm not really sure what's going on here... Some more comments would be nice.